src/Application/Kernel.php

```diff
@@ -161,4 +149,57 @@ private function initDiscovery(Container $container): void
             $discovery->storeCache();
         }
     }
+
+    private function discoverInstalledPackageLocations(): void
+    {
+        $composerPath = path($this->root, 'vendor/composer');
+        $installed = path($composerPath, 'installed.json');
+
+        if (is_file($installed)) {
+            $installed = $this->loadJsonFile($installed);
+
+            foreach ($installed['packages'] ?? [] as $package) {
+                $packagePath = realpath(path($composerPath, $package['install-path']));
```

Comment:
Why wrapping in `realpath`? If this is necessary, we might need to add it to the `path` helper function so that it is consistent.

Reply:
i wouldn't mind seeing it added to `path()` -- i get nervous with relative path segments, and when you try to read them in output or debugging it's easy to get confused. i prefer to have the resolved actual path whenever possible.

Your call: i can make the change to `path()` or remove `realpath()` here.

Reply:
Hm. `realpath()` fails on glob patterns like `.../tempest-framework/src/Config/**.php` (coming from `Kernel::initConfig()`

That's unfortunate.

Reply:
Oh, that's unfortunate indeed. How about we just use `path` as is for now, and we tackle `realpath` problems in a separate PR or issue after this PR gets merged. I like it when PRs focus on one core problem and not diverge too much from the initial goal :)

Reply:
Roger that. It could be as simple as `return realpath($path) ?: $path;` but i don't feel great about "your input has been changed, maybe" results.